app/views/conference/_conference_details.html.haml

```diff
@@ -26,8 +26,12 @@
             - if conference.registration_open?
               - if conference.user_registered?(current_user)
                 = link_to "My Registration", conference_conference_registrations_path(conference.short_title), :class =>"btn btn-default"
-              - else
+              - elsif !conference.registration_limit_exceeded?
                 = link_to "Register", new_conference_conference_registrations_path(conference.short_title), :class =>"btn btn-default"
```

Comment:
You could simply add at the end of this line:

disabled: registration_limit_exceeded?

And move the if statement on the next line so that you show the message, if needed.